### PR TITLE
OvS DPDK dataplane deployment and nodeset sample CR files

### PR DESCRIPTION
**Pull request by Jaganathancse**
Created Sample CR files for OpenstackDataPlaneDeployment and OpenstackDataPlaneNodeSet.

### CODE REVIEW

**Comment by Jaganathancse** [comment]
recheck

**Comment by Jaganathancse** [comment]
recheck

**Comment by Jaganathancse** [comment]
recheck

**Comment by Jaganathancse** [comment]
recheck

File: config/samples/dataplane_v1beta1_openstackdataplanenodeset_ovs_dpdk.yaml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+         edpm_nodes_validation_validate_gateway_icmp: false
+         # edpm nfv ovs dpdk config
+         edpm_kernel_args: "default_hugepagesz=1GB hugepagesz=1G hugepages=64 iommu=pt intel_iommu=on tsx=off isolcpus=2-11,14-23"
+         edpm_tuned_profile: "cpu-partitioning"
+         edpm_nova_libvirt_qemu_group: "hugetlbfs"
+         edpm_tuned_isolated_cores: "2-11,14-23"
+         edpm_ovn_dpdk_pmd_core_list: "1,13,2,14,3,15"
+         edpm_ovn_dpdk_socket_memory: "4096"
+         edpm_ovn_dpdk_memory_channels: "4"
+         edpm_ovn_dpdk_vhost_postcopy_support: "true"
```
Comment (jpodivin) [inline config/samples/dataplane_v1beta1_openstackdataplanenodeset_ovs_dpdk.yaml:137]: ditto

**Review by jpodivin — COMMENTED** [review]
There are some variables set that do not exist in the current RHOSO iteration. Furthermore the samples could use some documentation, either here, or in a linked PR somewhere else.

File: config/samples/dataplane_v1beta1_openstackdataplanenodeset_ovs_dpdk.yaml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+         edpm_nodes_validation_validate_controllers_icmp: false
+         edpm_nodes_validation_validate_gateway_icmp: false
+         # edpm nfv ovs dpdk config
+         edpm_kernel_args: "default_hugepagesz=1GB hugepagesz=1G hugepages=64 iommu=pt intel_iommu=on tsx=off isolcpus=2-11,14-23"
+         edpm_tuned_profile: "cpu-partitioning"
+         edpm_nova_libvirt_qemu_group: "hugetlbfs"
+         edpm_tuned_isolated_cores: "2-11,14-23"
+         edpm_ovn_dpdk_pmd_core_list: "1,13,2,14,3,15"
+         edpm_ovn_dpdk_socket_memory: "4096"
+         edpm_ovn_dpdk_memory_channels: "4"
```
[the remark this answers is not in the document]
Comment (Jaganathancse) [reply config/samples/dataplane_v1beta1_openstackdataplanenodeset_ovs_dpdk.yaml:136]: same

File: config/samples/dataplane_v1beta1_openstackdataplanenodeset_ovs_dpdk.yaml [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+         # edpm_nodes_validation
+         edpm_nodes_validation_validate_controllers_icmp: false
+         edpm_nodes_validation_validate_gateway_icmp: false
+         # edpm nfv ovs dpdk config
+         edpm_kernel_args: "default_hugepagesz=1GB hugepagesz=1G hugepages=64 iommu=pt intel_iommu=on tsx=off isolcpus=2-11,14-23"
+         edpm_tuned_profile: "cpu-partitioning"
+         edpm_nova_libvirt_qemu_group: "hugetlbfs"
+         edpm_tuned_isolated_cores: "2-11,14-23"
+         edpm_ovn_dpdk_pmd_core_list: "1,13,2,14,3,15"
+         edpm_ovn_dpdk_socket_memory: "4096"
```
[the remark this answers is not in the document]
Comment (Jaganathancse) [reply config/samples/dataplane_v1beta1_openstackdataplanenodeset_ovs_dpdk.yaml:135]: these are OvS DPDK rellated variables and already available in upstream.
https://github.com/openstack-k8s-operators/architecture/pull/80

File: config/samples/dataplane_v1beta1_openstackdataplanenodeset_ovs_dpdk.yaml (hunk 1 above)
Comment (Jaganathancse) [reply config/samples/dataplane_v1beta1_openstackdataplanenodeset_ovs_dpdk.yaml:137]: same

File: config/samples/dataplane_v1beta1_openstackdataplanenodeset_ovs_dpdk.yaml [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+                - type: ovs_dpdk_port
+                  name: dpdk2
+                  members:
+                  - type: interface
+                    name: nic4
+
+         # These vars are for the network config templates themselves and are
+         # considered EDPM network defaults.
+         neutron_physical_bridge_name: br-ex
+         neutron_public_interface_name: eth0
```
[the remark this answers is not in the document]
Comment (Jaganathancse) [reply config/samples/dataplane_v1beta1_openstackdataplanenodeset_ovs_dpdk.yaml:125]: used this config sample based on https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/config/samples/dataplane_v1beta1_openstackdataplanenodeset_baremetal_with_ipam.yaml
also available in other few sample files.

[the remark this answers is not in the document]
Comment (jpodivin) [reply config/samples/dataplane_v1beta1_openstackdataplanenodeset_ovs_dpdk.yaml:125]: That doesn't mean it's actually used by a role. Only that it's filled.

**Review by jpodivin — COMMENTED** [review]
Several parameters used in this sample are not actually used by edpm-ansible roles. They need to be either renamed, or the roles in edpm-ansible adjusted.

**Review by jpodivin — CHANGES_REQUESTED** [review]
Several parameters used in this sample are not actually used by edpm-ansible roles. They need to be either renamed, or the roles in edpm-ansible adjusted.

**Comment by Jaganathancse** [comment]
> Several parameters used in this sample are not actually used by edpm-ansible roles. They need to be either renamed, or the roles in edpm-ansible adjusted.

These parameters are specific to OvS DPDK (role edpm_ovs_dpdk) and not common.
Also this Dataplane CR is the sample  to enable OvS DPDK on compute nodes similar like SRIOV.
will discuss and do a seperate PR for naming changes.

**Review by jpodivin — APPROVED** [review]

**Review by fao89 — APPROVED** [review]

**Pull request merged**